black.py

```diff
@@ -1487,10 +1487,40 @@ def maybe_remove_trailing_comma(self, closing: Leaf) -> bool:
             )
             and closing.type == token.RPAR
         ):
+            self.remove_trailing_comma()
+            return True
+
+        # Otherwise, if the trailing one is the only one, we might mistakenly
+        # change a tuple into a different type by removing the comma.
+        if hasattr(closing, "bracket_depth"):
```

Comment:
Not sure when `closing` does not have the `bracket_depth` (and `opening_bracket`, see below) properties but test would fail with "Leaf has no `bracket_depth`/`opening_bracket` attribute" if I didn't add this